File: java-playground/src/main/java/org/gonza/javaplayground/util/RandomNumberGenerator.java
package org.gonza.javaplayground.util;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class RandomNumberGenerator {
	public static List<Integer> generate(int size, int start, int end) {
		Set<Integer> set = new HashSet<>();

		while (set.size() < size) {
			double d = Math.random() * end + start;
			set.add((int)d);
		}

		return new ArrayList<>(set);
	}
}

File: kotlin-playground/src/main/kotlin/org/gonza/kotlinplayground/domain/RandomNumberGenerator.kt
package org.gonza.kotlinplayground.domain

import java.util.Random

private const val DEFAULT_RANGE = 10

class RandomNumberGenerator(
    override val range: Int = DEFAULT_RANGE,
) : NumberGenerator {
    private val random = Random()

    override fun generate(): Int = random.nextInt(range)
}
